controllers/user/index.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+const express = require('express');
+const router = express.Router();
+
+const routeStatic = require('../utils/routeStatic');
+const auth = require('../../modules/authed');
+const register = require('./register');
+const login = require('./login');
+const password = require('./password');
+const { userBasePath, userEndpoints, makeUserLink, emailPasswordValidations, formValidations } = require('./utils');
+
+// serve a logged-in user's dashboard
+routeStatic.injectUrlRequest(router, userEndpoints.dashboard);
```

Comment:
More a question for another day but should `injectUrlRequest` live under a different namespace than `routeStatic`? This confused me a little as this step doesn't seem to have much to do with static pages. Do all routes need to call this `injectUrlRequest` method?

Reply:
Yeah, this is a bit ugly isn't it. Definite code smell.

Basically, templates that need access to `request` data (eg. for form input values or flash data) have it injected via `routeStatic` – there was a nasty race condition when I tried just setting an app-wide global called `request` (eg. opening lots of pages at once saw them override one another). Non-static routes (like these ones) don't have that variable automatically, so this method injects it.

But yeah, it feels a bit broken doesn't it. Will have a think...